src/modules/package.rs

```diff
@@ -184,80 +168,88 @@ fn extract_maven_version(file_contents: &str) -> Option<String> {
+    if !context
         .try_begin_scan()?
         .set_extensions(&["nimble"])
         .is_match()
     {
-        extract_nimble_version(context)
-    } else if let Ok(package_json) = utils::read_file(base_dir.join("package.json")) {
-        extract_package_version(&package_json, config.display_private)
-    } else if let Ok(poetry_toml) = utils::read_file(base_dir.join("pyproject.toml")) {
-        extract_poetry_version(&poetry_toml)
-    } else if let Ok(setup_cfg) = utils::read_file(base_dir.join("setup.cfg")) {
-        extract_setup_cfg_version(&setup_cfg)
-    } else if let Ok(composer_json) = utils::read_file(base_dir.join("composer.json")) {
-        extract_composer_version(&composer_json)
-    } else if let Ok(build_gradle) = utils::read_file(base_dir.join("build.gradle")) {
-        extract_gradle_version(&build_gradle)
-    } else if let Ok(project_toml) = utils::read_file(base_dir.join("Project.toml")) {
-        extract_project_version(&project_toml)
-    } else if let Ok(mix_file) = utils::read_file(base_dir.join("mix.exs")) {
-        extract_mix_version(&mix_file)
-    } else if let Ok(chart_file) = utils::read_file(base_dir.join("Chart.yaml")) {
-        extract_helm_package_version(&chart_file)
-    } else if let Ok(pom_file) = utils::read_file(base_dir.join("pom.xml")) {
-        extract_maven_version(&pom_file)
-    } else if let Ok(meson_build) = utils::read_file(base_dir.join("meson.build")) {
-        extract_meson_version(&meson_build)
-    } else if let Ok(vlang_mod) = utils::read_file(base_dir.join("v.mod")) {
-        extract_vmod_version(&vlang_mod)
-    } else if let Ok(vlang_vpkg) = utils::read_file(base_dir.join("vpkg.json")) {
-        extract_vpkg_version(&vlang_vpkg)
-    } else {
-        None
-    }
+        return None;
+    };
+
+    let cmd_output = context.exec_cmd("nimble", &["dump", "--json"])?;
+    let nimble_json: json::Value = json::from_str(&cmd_output.stdout).ok()?;
+
+    let raw_version = nimble_json.get("version")?.as_str()?;
+
+    format_version(raw_version, config.version_format)
 }
 
-fn format_version(version: &str) -> String {
-    let cleaned = version.replace('"', "").trim().to_string();
-    if cleaned.starts_with('v') {
-        cleaned
-    } else {
-        format!("v{}", cleaned)
-    }
+fn get_version(context: &Context, config: &PackageConfig) -> Option<String> {
+    let package_version_fn: Vec<fn(&Context, &PackageConfig) -> Option<String>> = vec![
+        get_cargo_version,
+        get_nimble_version,
+        get_node_package_version,
+        get_poetry_version,
+        get_setup_cfg_version,
+        get_composer_version,
+        get_gradle_version,
+        get_julia_project_version,
+        get_mix_version,
+        get_helm_package_version,
+        get_maven_version,
+        get_meson_version,
+        get_vmod_version,
+        get_vpkg_version,
+    ];
+
+    package_version_fn.iter().find_map(|f| f(context, config))
```

Comment:
Is there a reason `format_version` is called in each function instead of here? Beyond this LGTM.

Reply:
We do some matching on the parsed string when getting the version from the npm package.json and I couldn't come up with a way to make it work nicely.